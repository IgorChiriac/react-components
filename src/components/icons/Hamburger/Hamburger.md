This component is a wrapper around the svg code for Hamburger's icon.

    <Hamburger active size="30px" activeColor="blue" inactiveColor="gray" />
