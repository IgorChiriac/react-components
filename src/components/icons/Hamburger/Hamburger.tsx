import React from 'react';
import { colors } from '../../../constants/colors';

export interface IHamburgerProps {
  /**
   * The width and height of the SVG
   */
  size?: string;
  /**
   * Targets the SVG `fill` attribute
   */
  activeColor?: string;
  /**
   * Targets the SVG `fill` attribute
   */
  inactiveColor?: string;
  /**
   * For the default active state
   */
  active?: boolean;
}

export interface IHamburgerState {
  active: boolean;
}

const Hamburger: React.FunctionComponent<IHamburgerProps> = ({
  size = '40px',
  active = false,
  activeColor = colors.base.secondary,
  inactiveColor = colors.neutral.medium,
}) => {
  const color = active ? activeColor : inactiveColor;
  return (
    <svg width={size} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48">
      <path
        fill={color}
        d="M8 36h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2s.9 2 2 2zm0-10h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2s.9 2 2 2zM6 14c0 1.1.9 2 2 2h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2z"
      />
    </svg>
  );
};

export default Hamburger;
