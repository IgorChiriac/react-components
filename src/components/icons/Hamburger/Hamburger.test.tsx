import React from 'react';
import { render } from 'react-testing-library';
import Hamburger from './Hamburger';

describe('<Hamburger />', () => {
  it('renders the component with props', () => {
    const { container } = render(<Hamburger active size="24px" activeColor="black" inactiveColor="gray" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
