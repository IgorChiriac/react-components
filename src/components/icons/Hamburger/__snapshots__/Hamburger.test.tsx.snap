// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> renders the component with props 1`] = `
<svg
  viewBox="0 0 48 48"
  width="24px"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M8 36h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2s.9 2 2 2zm0-10h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2s.9 2 2 2zM6 14c0 1.1.9 2 2 2h32c1.1 0 2-.9 2-2s-.9-2-2-2H8c-1.1 0-2 .9-2 2z"
    fill="black"
  />
</svg>
`;
